### DISCUSSION TITLE
Permanent sessions False and session changes

### DISCUSSION THREAD

**Discussion by aktoak** [miguelgrinberg/Flask-SocketIO#1686] [Q&A] [answered]
i noticed when making flask session not permanent in the config file using `SESSION_PERMENANT = FALSE` opposed to the default value which is true.
then the changes made to the session from socket.io routes are not passed to the http routes?
when using permanent sessions everything works as expected.
is there a way to fix this? or maybe explain why this is happening? 

P.S i am using Flask-Sessions (server side sessions)

**Comment by miguelgrinberg** [comment]
Do you have example code? Without knowing how you configured Flask and Flask-Session it is not possible for me to debug. If at all possible, it would be useful if you can modify the sessions example in this repository to show the problem. Thanks.

**Comment by aktoak** [comment, marked as answer]
so after some research, turns out the problem was not with flask-socketio or flask-session but rather with flask-login, i had my 
`session_protection` set to `strong` which caused `current_user.is_authenticated` to return false after calling `login_user` from socketio route
changing the `session_protection` to `basic or None` solves this issue depending on the needs.

Thanks,
